core/src/main/java/org/apache/calcite/plan/RelTraitSet.java

```diff
@@ -417,13 +420,34 @@ public int size() {
    * @return true if traits are equal and in the same order, false otherwise
    */
   @Override public boolean equals(Object obj) {
-    return this == obj
-        || obj instanceof RelTraitSet
-        && Arrays.equals(traits, ((RelTraitSet) obj).traits);
+    if (this == obj) {
```

Comment:
Given that the RelTraitSet is always canonized, can we just `return this == obj;` here?